htof/parse.py

```diff
@@ -178,6 +181,108 @@ def __init__(self, scan_angle=None, epoch=None, residuals=None, inverse_covarian
+        fpath = f"HIP{star_id}.csv"
+        path = os.path.join(os.getcwd(), f"{intermediate_data_directory}/{fpath}")
+        os.makedirs(intermediate_data_directory, exist_ok=True)
+        data.to_csv(path, index=False, index_label=False)
+        return None
+
+    def query_gost_xml(self, target):
+        url = f"https://gaia.esac.esa.int/gost/GostServlet?name={target}&service=1"
+        try:
+            with requests.Session() as s:
+                s.get(url)
+                headers = {"Cookie": f"JSESSIONID={s.cookies.get_dict()['JSESSIONID']}"}
+                response = requests.request("GET", url, headers=headers)
+                return response.text
+        except:
+            warnings.warn("Querying the GOST service failed.")
+            return None
+
+    def parse_xml(self, response):
+        columns = ["Target", "ra[rad]", "dec[rad]", "ra[h:m:s]", "dec[d:m:s]", "ObservationTimeAtGaia[UTC]",
+                   "CcdRow[1-7]", "zetaFieldAngle[rad]", "scanAngle[rad]", "Fov[FovP=preceding/FovF=following]",
+                   "parallaxFactorAlongScan", "parallaxFactorAcrossScan", "ObservationTimeAtBarycentre[BarycentricJulianDateInTCB]"]
+        rows = []
+        root = ET.fromstring(response)
+        name = root.find('./targets/target/name').text
+        raR = root.find('./targets/target/coords/ra').text
+        decR = root.find('./targets/target/coords/dec').text
+        raH = root.find('./targets/target/coords/raHms').text
+        decH = root.find('./targets/target/coords/decDms').text
+        for event in root.findall('./targets/target/events/event'):
+            details = event.find('details')
+            observationTimeAtGaia = event.find('eventUtcDate').text
+            ccdRow = details.find('ccdRow').text
+            zetaFieldAngle = details.find('zetaFieldAngle').text
+            scanAngle = details.find('scanAngle').text
+            # TODO: FovP/FovF difference
+            fov = details.find('fov').text
+            parallaxFactorAl = details.find('parallaxFactorAl').text
+            parallaxFactorAc = details.find('parallaxFactorAc').text
+            observationTimeAtBarycentre = event.find('eventTcbBarycentricJulianDateAtBarycentre').text
+            rows.append([name, raR, decR, raH, decH, observationTimeAtGaia, ccdRow,
+                         zetaFieldAngle, scanAngle, fov, parallaxFactorAl, parallaxFactorAc, observationTimeAtBarycentre])
+        data = pd.DataFrame(rows, columns=columns)
+        data = data.astype({"Target": str,"ra[rad]": float, "dec[rad]": float,"ra[h:m:s]": str,"dec[d:m:s]": str,"ObservationTimeAtGaia[UTC]": str,"CcdRow[1-7]": int,"zetaFieldAngle[rad]": float,"scanAngle[rad]": float,"Fov[FovP=preceding/FovF=following]": str,"parallaxFactorAlongScan": float,"parallaxFactorAcrossScan": float,"ObservationTimeAtBarycentre[BarycentricJulianDateInTCB]": float })
+        return data
+    
+    def keep_field_hits(self, data):
+        format = "%Y-%m-%dT%H:%M:%S.%f"
```

Comment:
This function would benefit from a doc string. Something like 
"""
Gost files downloaded from the web contain sequences of ten observations, for every observation of the star in the scanning law. The first entry is the skymapper CCD hit, and the extra entries are redundant (the hits for astrometric field CCD's 1 through 9). Only the second observation of each sequence should be saved. This function saves the second observation (this is the hit on the first astrometric field CCD (AF1)).
"""

Reply:
I agree, I have added the doc string.